File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        vector<int> vis(256,-1);
        int ans=0;
        int l=0;
        for(int r=0;r<s.size();r++){
            if(vis[s[r]] >= l) l=vis[s[r]]+1;
            vis[s[r]] = r;
            if(r-l+1 > ans) ans=r-l+1;
        }
        return ans;
    }
};

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int ans =0 ;
        int[] vis = new int[256];
        Arrays.fill(vis,-1);
        int l=0;
        for(int i=0;i<s.length();i++){
            if(vis[s.charAt(i)] >= l) {
                l = vis[s.charAt(i)] + 1; 
            }
            vis[s.charAt(i)] = i;
            if(i-l+1 > ans) ans = i-l+1;
        }
        return ans;
    }
}